pkgs/test/lib/src/runner/browser/dom.dart

```diff
@@ -2,29 +2,74 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:js_interop';
 import 'dart:js_util' as js_util;
 
 import 'package:js/js.dart';
 
+// Conversion functions to help with the migration.
+extension ObjectToJSAnyExtension on Object? {
+  JSAny? get toJSAnyShallow {
+    if (const bool.fromEnvironment('dart.library.html')) {
+      // Cast is necessary on Wasm backends and in the future on JS backends as
+      // well.
+      // ignore: unnecessary_cast
+      return this as JSAny?;
```

Comment:
Is there an issue that should be filed? Will this warn in the compilers as well?

Reply:
This cast will no longer be necessary after https://dart-review.googlesource.com/c/sdk/+/283742 lands. However, that CL is currently blocked on front-end work.

The root problem is that JS types are reified on Wasm backends, but not yet reified on JS backends. On JS backends, JS types are just typedefs for Dart types, hence the cast warning.

It may be that this CL lands after the reification CL, so we may not need this cast after all. If not, we can file an issue to track this if you like. Thoughts?